### PR TITLE
Implement PS-2477 (Basic test cases should be created for PAM plugin)

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
Although PAM authentication is impossible to test in MTR without
system cooperation, it is possible to do minimal plugin testing
without testing authentication itself: that INSTALL/UNINSTALL PLUGIN
works and that it is possible to CREATE USER with this authentication
method. Add such tests to the plugin-specific test directory, add this
suite to the default suite list, and install the test files.

https://ps.cd.percona.com/view/5.6/job/percona-server-5.6-param/15/ (tests actually pass, red dots is a Jenkins issue)

### CODE REVIEW

File: mysql-test/include/have_auth_pam_compat.inc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#
+# Check if server has support for loading plugins
+#
+if (`SELECT @@have_dynamic_loading != 'YES'`) {
+  --skip Example plugin requires dynamic loading
```
Comment (dutow) [inline mysql-test/include/have_auth_pam_compat.inc]: "Example plugin"?

File: mysql-test/include/have_auth_pam.inc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#
+# Check if server has support for loading plugins
+#
+if (`SELECT @@have_dynamic_loading != 'YES'`) {
+  --skip Example plugin requires dynamic loading
```
Comment (dutow) [inline mysql-test/include/have_auth_pam.inc]: "Example plugin?"

**Review by percona-ysorokin — APPROVED** [review]
LGTM

File: plugin/percona-pam-for-mysql/tests/mtr/auth_pam_compat-master.opt [hunk 3]
```diff
@@ -0,0 +1 @@
+$AUTH_PAM_OPT
```
Comment (dutow) [inline plugin/percona-pam-for-mysql/tests/mtr/auth_pam_compat-master.opt]: AUTH_PAM_COMPAT_OPT?

**Comment by percona-ysorokin** [comment]
It looks like I missed everything :)

**Comment by laurynas-biveinis** [comment]
Fixed

**Pull request merged**